### ISSUE TITLE
ValidationPipe's enableImplicitConversion fails with Mapped Types

### ISSUE THREAD

**Issue by liadmat** [nestjs/nest#6794]
## Current behavior
When using `ValidationPipe` with `enableImplicitConversion`, implicit conversion doesn't happen for DTOs using Mapped Types from `@nestjs/swagger` or `@nestjs/mapped-types` (e.g. `PickType`).

## Input Code
**Minimal reproduction**: https://github.com/liadmat/nestjs-repo

1. Run `npm start`
2. Visit http://localhost:3000/todos/123.
3. See the following response:
  ```
  {"statusCode":400,"message":["id must be a number conforming to the specified constraints"],"error":"Bad Request"}
  ```
  
  Which is unexpected, because `enableImplicitConversion` is turned on.
  
4. Comment out the current implementation of `find-one-todo-params.dto.ts`, and uncomment the other implementation.
5. Visit http://localhost:3000/todos/123 once again.
6. See the following response, which is expected:
  ```
  ID: 123
  ```

## Expected behavior

I'd expect the 2 versions to be 100% equivalent.

## Possible Solution

No suggestions, but perhaps the `design:types` gets lost somewhere along the way.

## Environment

<pre><code>
Nest version: 7.5.6

</code></pre>

**Comment by jmcdo29** [comment]
I can reproduce this. If the following is added to the `bootstrap` in the `main.ts` you can see that the original entity keeps its `deign:type` metadata, but the `FindOneTodoParams` loses the `deign:type` metadata.

```ts
console.log(Reflect.getMetadata('design:type', Todo.prototype, 'id'));

  console.log(
    Reflect.getMetadata('design:type', FindOneTodoParams.prototype, 'id'),
  );
```
```
[Nest] 76646   - 03/29/2021, 2:03:15 PM   [NestFactory] Starting Nest application...
[Nest] 76646   - 03/29/2021, 2:03:15 PM   [InstanceLoader] AppModule dependencies initialized +26ms
[Function: Number]
undefined
[Nest] 76646   - 03/29/2021, 2:03:15 PM   [RoutesResolver] TodosController {/todos}: +6ms
[Nest] 76646   - 03/29/2021, 2:03:15 PM   [RouterExplorer] Mapped {/todos/:id, GET} route +2ms
[Nest] 76646   - 03/29/2021, 2:03:15 PM   [NestApplication] Nest application successfully started +1ms
```

**Comment by kamilmysliwiec** [comment]
Fixed in `@nestjs/mapped-types@0.4.1`

**Comment by liadmat** [comment]
Thanks for the quick fix @kamilmysliwiec ! Can we please get this dependency updated in `@nestjs/swagger`?
